encoding/thrift/options.go

```diff
@@ -154,3 +156,21 @@ func (p protocolOption) applyRegisterOption(c *registerConfig) {
 func Protocol(p protocol.Protocol) Option {
 	return protocolOption{Protocol: p}
 }
+
+// NoWire is an option that specifies to *not* use the thriftrw.Wire
+// intermediary format.  Note that if this is enabled, and a
+// 'protocol.Protocol' is provided (e.g., via thrift.Protocol)m that provided
+// protocol must be able to satisfy the 'stream.Protocol' interface.
+func NoWire(enable bool) Option {
```

Comment:
do we want to take a bool here? If no one calls this options, you have NoWire set to zero value false. 
`func NoWire() Option {return noWireOption{Enable: true}}` is good enough?

Reply:
This is set this way in case we decide to make the "NoWire" option the default, leaving this with an argument allows users to disable it if they wish to.

Reply:
@AllenLuUber, with functional options, there's an advantage to the bool argument.
Specifically, with regards to dynamically constructing options or wrapping functions.

```
func foo(opts ...Option) {
  var finalOpts []Option
  finalOpts = append(finalOpts, Bar()) // default options
  finalOpts = append(finalOpts, opts...) // user options
  anotherFunc(opts...)
}
```

If `Bar()` takes no parameters, like a boolean CLI flag, then it becomes impossible to disable it if a prior `...Option` list enabled it.

This actually bit us in Zap's [AddCaller()](https://pkg.go.dev/go.uber.org/zap#AddCaller) option that takes no parameters, and we had to eventually add [WithCaller(bool)](https://pkg.go.dev/go.uber.org/zap#WithCaller) which allows enabling or disabling regardless of prior choice.

Reply:
@witriew May or may not be worth calling out that in the future this will become the default. Up to you.